PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/ruleConfigs.json

```diff
@@ -10,5 +10,16 @@
   "ServiceBusReceiverAsyncClientCheck": {
     "client_name": "ServiceBusReceiverAsyncClient",
     "antipattern_message": "Use of ServiceBusReceiverAsyncClient detected. Use ServiceBusProcessorClient instead."
+  },
+  "KustoQueriesWithTimeIntervalInQueryStringCheck": {
+    "regex_patterns": {
+      "KQL_ANTI_PATTERN_AGO": "\"?ago\\(\"?\\+?\"?\\d+[smhdw]\"?\\+?\"?\\)?\"?",
+      "KQL_ANTI_PATTERN_DATETIME": ".*datetime\\s*\\(\\s*[\"\\+\\s]*\\d{4}-\\d{2}-\\d{2}(?:T\\d{2}:\\d{2}:\\d{2}Z)?[\"\\+\\s]*\\s*\\)",
```

Comment:
These patterns won't work if the user uses variables to do string concatenation.

```java
String startDate = "";
String endDate = "";

String kustoQuery = "Requests | where requestTime between (" + startDate + " .. " + endDate + ")";
```

Reply:
yes I check for both literal expressions and operands -- the handlePolyadicExpression(PsiPolyadicExpression polyadicExpression) method replaces the String variable name with its values to make a complete literal expression that the regex is then checked against

Reply:
I updated the PR screenshots to show this

Reply:
The variable replacement may not work if the string values are not literals. For e.g. `String startDate = getStartDateFromUser()` will be resolved to a datetime only at runtime when the user actually selects a date. So, we won't be able to warn at compile time.

Reply:
check for operators of datetime expressions(between, ago and the patterns it exists in). still need regex to find operators, not necessarily the value of the datetime